### PR TITLE
CTC-59: browse data for media clips page

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: [2017](https://www.youtube.com/watch?v=_NSuIYwBxu4)

Teacher media clip pages now render their data from a query requests rather than hardcoded data on the page

- Creates `lessonMediaClips` directory with query, test files, schemas and fixture data
- Creates construct functions for both canonical and canonical lesson media clips 
- Adds these queries to the media pages so they render lesson data via the `getStaticProps` method

## Issue(s)

Fixes #CTC-59

### Acceptance Criteria

- [ ]  A stub query is created for media clips page
- [ ]  browse data returned from getStaticProps
- [ ]  breadcrumbs implemented
- [ ]  handles canonical and browse versions

## How to test

1. Go to https://deploy-preview-3013--oak-web-application.netlify.thenational.academy/teachers/programmes/chemistry-secondary-ks4-foundation-edexcel/units/atomic-structure-and-the-periodic-table/lessons/atomic-structure-negligible-electron-mass/media
https://deploy-preview-3013--oak-web-application.netlify.thenational.academy/teachers/lessons/atomic-structure-negligible-electron-mass/media
 - or append /media to any lesson page
2. You should see a media page with a working breadcrumb component

## Screenshots

How it should now look:
{screenshots}
#### Canonical Lesson
Screenshot 2024-11-22 at 16 46 29

#### Non-canonical Lesson
Screenshot 2024-11-22 at 16 46 45

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/node-lib/curriculum-api-2023/queries/lessonMediaClips/lessonMediaClips.query.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      // do the same for features
+      res.browseData = res.browseData.map((bd) => ({
+        ...bd,
+        features: featuresIntersection,
+      }));
+    }
+
+    /**
+     * TODO: Add media clip query name to curriculum schema
+     */
```
Comment (k-huggs) [inline src/node-lib/curriculum-api-2023/queries/lessonMediaClips/lessonMediaClips.query.ts:98]: https://github.com/oaknational/oak-curriculum-schema/pull/66

**Comment by RBrownsett** [comment]
@k-huggs just checking, when I open both links (and any other lesson with `/media` appended) I briefly see a 404 page. Is this expected as the page is built? 

https://github.com/user-attachments/assets/a547ba2c-0aa3-4992-ae04-0d054ee2c957

**Review by RBrownsett — APPROVED** [review]
Approved - tested. Breadcrumbs working as expected.

File: src/node-lib/curriculum-api-2023/queries/unitListing/filters/getAllCategories.ts [hunk 2]
```diff
@@ -45,7 +45,8 @@ const generateCategoryObjects = (uniqueCategories: Set<string>): Category[] => {
     const categoryKey = category as CategoryKeys;
     categories.push({
       label: category,
-      iconName: categoryIconMap[categoryKey],
+      // default to subject-english icon if no icon is found
+      iconName: categoryIconMap[categoryKey] || "subject-english",
```
Comment (weronika-szalas) [inline src/node-lib/curriculum-api-2023/queries/unitListing/filters/getAllCategories.ts:49]: what is the reason behind this?

**Review by weronika-szalas — APPROVED** [review]

**Pull request merged**